languages/csharp/AppEncryption/README.md

```diff
@@ -224,63 +95,146 @@ Dictionary<string, string> regionDictionary = new Dictionary<string, string>
+
+```c#
+AppEncryption<byte[], byte[]> appEncryptionBytes = appEncryptionSessionFactory.GetAppEncryptionBytes("some_user");
+```
+
+The different usage styles are explained below.
+
+**NOTE:** Remember to close the `AppEncryption` session after all cryptographic operations to dispose of associated resources.
+
+#### Plain Encrypt/Decrypt Style
+This usage style is similar to common encryption utilities where payloads are simply encrypted and decrypted, and it is 
+completely up to the calling application for storage responsibility.
+
+```c#
+string originalPayloadString = "mysupersecretpayload";
+
+// encrypt the payload
+byte[] dataRowRecordBytes = appEncryptionBytes.Encrypt(Encoding.UTF8.GetBytes(originalPayloadString));
+
+// decrypt the payload
+string decryptedPayloadString = Encoding.UTF8.GetString(appEncryptionBytes.Decrypt(newDataRowRecordBytes));
+```
+
+#### Custom Persistence via Store/Load methods
+Asherah supports a key-value/document storage model. An [AppEncryption](AppEncryption/AppEncryption.cs) instance can 
+accept a [Persistence](AppEncryption/Persistence/Persistence.cs) implementation and hook into its `Load` and `Store` 
+calls.
+
+An example `Dictionary`-backed `Persistence` implementation:
+
+```c#
+public class DictionaryPersistence : Persistence<JObject>
+{
+    Dictionary<string, JObject> dictionaryPersistence = new Dictionary<string, JObject>();
 
-    // ...
+    public override Option<JObject> Load(String key)
+    {
+        return dictionaryPersistence.TryGetValue(key, out JObject result) ? result : Option<JObject>.None;
+    }
+
+    public override void Store(String key, JObject value)
+    {
+        dictionaryPersistence.Add(key, value);
+    }
 }
 ```
-To report metrics to AmazonCloudWatch please refer to AppMetricsCloudWatchReporter (add link when we open source it).
+
+An example end-to-end use of the store and load calls:
+
+```c#
+// Encrypts the payload, stores it in the dictionaryPersistence and returns a look up key
+string persistenceKey = appEncryptionJsonImpl.Store(originalPayload.ToJObject(), dictionaryPersistence);
+
+// Uses the persistenceKey to look-up the payload in the dictionaryPersistence, decrypts the payload if any and then returns it
+Option<JObject> payload = appEncryptionJsonImpl.Load(persistenceKey, dictionaryPersistence);
+```
 
 ## Deployment Notes
-Please refer to the [Java lib's notes](../../java/app-encryption#deployment-notes) until documentation cleanup/refactor is complete.
 
-## Library Development Notes
+### Handling read-only Docker containers
 
-### Running Tests Locally via Docker Image
-Below is an example of how to run tests locally using a Docker image. This one is using the build image used for Jenkins build/deployment, but could be replaced with other images for different targeted platforms. Note this is run from your project directory.
+Dotnet enables debugging and profiling by default causing some system level writes. Disabling them ensures that the 
```

Comment:
- `system level writes` is a bit ambiguous. suggest `writes to the `filesystem` or `filesystem writes`
- nit: can probably get rid of `some`

may as well make similar changes in the reference app's Dockerfile comments where this came from